jiant/scripts/postproc/xtreme/xtreme_submission.py

```diff
@@ -0,0 +1,413 @@
+    elif phase == "test":
+        task_name_list = runner.jiant_task_container.task_run_config.test_task_list
+    else:
+        raise KeyError(phase)
+    task_name_list = [task_name for task_name in task_name_list if task_name.startswith(supertask)]
+    if phase == "val":
+        test_results_dict = runner.run_val(task_name_list=task_name_list)
+    elif phase == "test":
+        test_results_dict = {}
+        test_dataloader_dict = runner.get_test_dataloader_dict()
+        for task_name in task_name_list:
+            test_results_dict[task_name] = mock_qa_run_test(
+                test_dataloader=test_dataloader_dict[task_name],
+                jiant_model=runner.jiant_model,
+                task=runner.jiant_task_container.task_dict[task_name],
+                device=runner.device,
+                local_rank=runner.rparams.local_rank,
+                verbose=True,
+            )
+    else:
+        raise KeyError(phase)
+
+    # Generate QA preds
+    tokenizer = runner.model.tokenizer
+    for task_name in task_name_list:
+        task_results = test_results_dict[task_name]
+        task = runner.jiant_task_container.task_dict[task_name]
+        logits = task_results["accumulator"].get_accumulated()
+        lang = get_qa_language(supertask=supertask, task=task)
+        if phase == "val":
+            cached = runner.get_val_dataloader_dict([task_name])[
+                task_name
+            ].dataset.chunked_file_data_cache.get_all()
+        elif phase == "test":
+            cached = runner.get_test_dataloader_dict()[
+                task_name
+            ].dataset.chunked_file_data_cache.get_all()
+        else:
+            raise KeyError(phase)
+        data_rows = [row["data_row"] for row in cached]
+        results, predictions = squad_lib.compute_predictions_logits_v3(
+            data_rows=data_rows,
+            logits=logits,
+            n_best_size=task.n_best_size,
+            max_answer_length=task.max_answer_length,
+            do_lower_case=model_resolution.resolve_is_lower_case(tokenizer),
+            version_2_with_negative=task.version_2_with_negative,
+            null_score_diff_threshold=task.null_score_diff_threshold,
+            skip_get_final_text=(lang == "zh"),
+            tokenizer=tokenizer,
+        )
+        test_results_dict[task_name]["preds"] = predictions
+
+    jiant_evaluate.write_preds(
+        eval_results_dict=test_results_dict, path=preds_pickle_path,
+    )
+    preds_output_dir = os.path.join(output_dir, "preds", supertask)
+    os.makedirs(preds_output_dir, exist_ok=True)
+    for task_name, task_results in test_results_dict.items():
+        task = runner.jiant_task_container.task_dict[task_name]
+        lang = get_qa_language(supertask=supertask, task=task)
+        py_io.write_json(task_results["preds"], os.path.join(preds_output_dir, f"test-{lang}.json"))
+    print(f"Wrote {supertask} preds for {len(test_results_dict)} languages")
+
+
+def mock_qa_run_test(
```

Comment:
What does this function do?

Reply:
Turns out it's just `run_test` except I don't want to call `get_preds_from_accumulator`. Refactored.